cdap-ui/app/cdap/components/VirtualScroll/index.tsx

```diff
@@ -0,0 +1,123 @@
+/*
```

Comment:
Not as of now. It will take some substantial feature parity to see how this implementation fares. But IMHO we will not be on-par,  at least for sometime, with the performance offered by those libraries developed in open with 100s of contributions.

Reply:
Is there a reason for creating a new one then?

Reply:
Couple of things,
1. Historically we usually use a library for everything we want to do for anything other than basic rendering. This is an effort towards minimizing importing a whole slew of dependency and increase yarn install time (Today it takes > 5 mins to do yarn install)
2. I may have miscommunicated there, by performance I mean feature parity. The performance, in terms of the ability to render things efficiently is still the core objective and we can test and compare those numbers but the comparison would be limited to rendering a list or table and not a grid.
3. The flexibility to use web workers as part of the virtual scroll component for some of the processing that we might have to do. For instance sorting/searching large list or doing a compute intensive operation (parse json for valid avro schema).

It all boils down to whether we want to use industry famous libraries and incur the dependency and learning cost or develop it in-house for greater flexibility on the source.

Reply:
That makes sense. I agree it's a tough trade-off.